docs/source/API/core/MultiGPUSupport.rst

```diff
@@ -0,0 +1,153 @@
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
+Multi-GPU Support
+=================
+
+Kokkos has support for launching kernels on multiple GPU devices within a single node from host on a single MPI rank for the CUDA, HIP, and SYCL backends.
+
+Using this feature requires knowledge of backend specific API calls for creating non-default execution space instances. Once the execution space has been
+created, it can be used to create execution policies, allocate views, etc. all on the device chosen by the user.
+
+Constructing Execution Spaces
+-----------------------------
+
+CUDA
+~~~~
+
+For CUDA backend, the user creates a ``cudaStream_t`` object and passes it to the ``Kokkos::Cuda`` constructor.
+
+.. note:: The user is expected to manage the lifetime of the stream. Only after the execution space is destroyed
+          can the user safely destroy the stream.
+
+
+.. code-block:: cpp
+
+    // Query number of devices available
+    int n_devices;
+    cudaGetDeviceCount(&n_devices);
+
+    // Choose some 0 <= N < n_devices
+    int N = ...;
+
+    // Create stream on device N
+    cudaStream_t stream;
+    cudaSetDevice(N);
+    cudaStreamCreate(&stream);
+
+    // Scope execution space to ensure stream
+    // is destroyed *after* execution space
+    {
+      // Create execution space
+      Kokkos::Cuda exec_space(stream);
+
+      // Use execution space
+      /* ... */
+    }
```

Comment:
When do we initialize? Since objects are destroyed in the reverse order of creation, we would not need a scope to enforce that, right?

Reply:
We have to destroy the stream below before the execution space would go out of scope, assuming they are in the same function.

Do you mean initialize kokkos? We could do that any time before creating the execution space.

This is how the tutorial looks 
```cpp

cudaStreamCreate(&stream);

Kokkos::initialize(...);
{
  Kokkos::Cuda exec_space(stream);

  ...
}
Kokkos::finalize();

cudaStreamDestroy(stream);
```
I could be misunderstanding the question.

Reply:
It's not because of `stream` being deleted after the execution space, it's that we're required to manually destroy the stream before it's deleted.

But maybe then it wouldn't matter if we call `cudaStreamDestroy(stream)` even while the execution space existed? It would be awkward though if we had an execution space instance floating around with an invalid stream.

Reply:
oh, ok. "Lifetime" is the word that triggered me then. For me "Lifetime" in c++ is the time from creation until destruction (destructor call) of the class instance. IIUC you mean the lifetime of the execution space needs to end before we destroy (aka invalidate) the stream before the lifetime of the stream can end. Is that right?

Reply:
Yeah, that's right!

Reply:
ok, in that case we definitely need the scope. Maybe we can make it more expressive, e.g.:

.. note:: The user needs to manually manage the lifetime of Kokkos objects that use the stream. The lifetime of all Kokkos objects including the execution space that are related to a stream needs to end before the stream is invalidated via `cudaStreamDestroy`. This can be done via adding a scope as done in the example below.

Reply:
Updated the note, let me know what you think.

Reply:
I like it